python/ray/tune/tune.py

```diff
@@ -698,22 +677,25 @@ class and registered trainables.
         local_path=local_path, remote_path=remote_path, sync_config=sync_config
     )
 
-    checkpoint_score_attr = checkpoint_score_attr or ""
+    checkpoint_config = checkpoint_config or CheckpointConfig()
+    checkpoint_score_attr = (
+        # Note(jungong) : we must use _tune_legacy_checkpoint_score_attr
+        # here, instead of checkpoint_score_attribute, to make sure we
+        # handle legacy cases correctly (e.g., someone purposefully
+        # specify checkpoint_score_order="min-loss").
+        checkpoint_config._tune_legacy_checkpoint_score_attr
+        or ""
+    )
     if checkpoint_score_attr.startswith("min-"):
-        checkpoint_score_attr = checkpoint_score_attr[4:]
-        checkpoint_score_order = "min"
+        logger.warning(
+            "using min- and max- prefixes to specify checkpoint score "
+            "order is deprecated. Please use CheckpointConfig.checkpoint_score_order "
+            "instead"
+        )
```

Comment:
We can use `warnings.warn(msg, DeprecationWarning)` instead for this actionable deprecation warning.

Reply:
done